### ISSUE TITLE
Strong validation on the timestamp when writing data into the TSDB 

### ISSUE THREAD

**Issue by spicoflorin** [OpenTSDB/opentsdb#572]
Hello!
  I have some use cases for writing the data into TSDB, that cannot be fulfilled with current version due to the strong validation on the timestamp in the class IncomingDataPoints.java. The condition for always writing new data prevents me to add data from the past.
// always maintain last_ts in milliseconds
    if ((ms_timestamp ? timestamp : timestamp * 1000) <= last_ts) {
      throw new IllegalArgumentException("New timestamp=" + timestamp
          + " is less than or equal to previous=" + last_ts
          + " when trying to add value=" + Arrays.toString(value)
          + " to " + this);
    }
The use cases are the following:
1. I have many sensors data file that contains values for a sensor that they are not ordered by timestamp. This is normal in real life. 
2. A client uploads file from with that sensor data from year 2009. After a while,
a new client uploads its file with the same sensors but from  year 2008. 
We would like to use OpenTSDB for these two scenarios, but the above limitations prevents us to use it.
Is this strong validation a core precondition for storing the data in TSDB?  What implications involve if you disable it? 
Is OpenTSDB suitable for my use cases?

**Comment by manolama** [comment]
Hiya,

The check for time ordering is only encountered when bulk loading via the import CLI or if you use it via the Java API. It's really there just as a check since most time series are ordered so that if a file contained something out of order, the user could be alerted.

But there isn't any particular constraint that really requires the check so you're perfectly safe disabling it and if you like, you could raise a PR with a config flag to disable the check.

Re 1) do you have a sensor that writes data out of order somehow? e.g.
```
sensor.temp t1 v1
sensor.temp t3 v3
sensor.temp t2 v2
```

That sounds kinda odd.

2) That should be fine as it is if you're calling import with different files in a different order. The CLI should only track ordering between runs if I recall correctly.

**Comment by spicoflorin** [comment]
Hi!
  Thanks for the reply. The time order validation is applied only when using the command line bulk loading. Is throwing and exception and the JVM is halted. The Java API via TSDB#addPoint works fine without validation. 
1. How can I disable the validation? Do you have an option for doing that or do I have to change the code?
2. It can happen that the time series for a sensor to be un-ordered in the files.
3. If I'm importing many files with the same import process, we should check if the validation occurs. From the code it seems that validation remains per running process. So if If I would like to import many files I should an import command per file. Not tested yes.

**Comment by manolama** [comment]
Hiya, For now the only option is to comment out that section of code and recompile, then you should be good. We can add an option in 2.3 to disable the validation. I'll tag this and leave it open for that. Thanks!

**Comment by bobrik** [comment]
+1 on optionally disabling this. I exported part of OpenTSDB to validate different export scenarios and have duplicate datapoints:

```
mymetric 1450106640 1 type=something
mymetric 1450106640 2 type=something
```

```
Exception in thread "main" java.lang.IllegalArgumentException: New timestamp=1450106640 is less than or equal to previous=1450106640000 when trying to add value=[2] to IncomingDataPoints([0, 14, 112, 86, 110, -39, 112, 0, 0, 5, 0, 67, 44] (metric=mymetric), base_time=1450105200 (Mon Dec 14 15:00:00 UTC 2015), [])
	at net.opentsdb.core.IncomingDataPoints.addPointInternal(IncomingDataPoints.java:279)
	at net.opentsdb.core.IncomingDataPoints.addPoint(IncomingDataPoints.java:364)
	at net.opentsdb.tools.TextImporter.importFile(TextImporter.java:194)
	at net.opentsdb.tools.TextImporter.main(TextImporter.java:76)
```

Also new timestamp is in seconds while previous is in milliseconds.

It's nice to have a metric name and value here, though (#664).

**Comment by kcech** [comment]
+1. I cannot import a file previously exported by the scan utility. For some reason, the exported datapoint list contains a couple of duplicated rows. This issue prevents me to import the metrics without having to manually deleting the duplicates.

**Comment by manolama** [comment]
Merged in 71f35ca4a4cc4763110ac10d8e9ff265e7d6a750 with 2.4 snapshot (since it's a new feature). Please try it out and let us know how it looks.
